aten/src/ATen/native/cpu/LerpKernel.cpp

```diff
@@ -9,7 +9,36 @@ namespace native {
 namespace {
 
 void lerp_scalar_kernel(at::TensorIteratorBase& iter, const Scalar& weight) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(iter.common_dtype(), "lerp_kernel_scalar", [&] {
+  if (iter.common_dtype() == kBFloat16) {
+    using bVec = Vectorized<BFloat16>;
+    using fVec = Vectorized<float>;
+    float weight_val = weight.to<float>();
+    auto weight_vec = fVec(weight_val);
+    auto threshold_vec = fVec(0.5);
+    auto one_vec = fVec(1);
+    at::native::cpu_kernel_vec(
+      iter,
+      [weight_val](BFloat16 self_val, BFloat16 end_val) -> BFloat16 {
+        return (weight_val < 0.5)
```

Comment:
Generally looks good to me. The bfloat16 is cast to float for scalar operation. It should have similar perf as float. But why in your perf table, the scalar bfloat16 is much faster than scalar float?

Reply:
Because bfloat16 uses cpu_kernel_vec and blendv, which is faster than cpu_kernel.